packages/ckeditor5-emoji/src/emojirepository.ts

```diff
@@ -108,7 +120,8 @@ export default class EmojiRepository extends Plugin {
 		// Store the emoji database after normalizing the raw data.
 		this._database = emojiDatabase
 			.filter( item => isEmojiCategoryAllowed( item ) )
-			.filter( item => EmojiRepository._isEmojiSupported( item, container ) )
+			.filter( item => item.version <= emojiSupportedVersionByOs )
+			.filter( item => !EmojiRepository._hasZwj( item.emoji ) ? true : EmojiRepository._isEmojiZwjSupported( item, container ) )
```

Comment:
These two `.filter()` calls do the same thing from a business perspective: filtering out unsupported emojis. Is it possible to merge this logic from both `.filter()` into one and hide the technical details in this place? Something like it was before:

```
.filter( item => EmojiRepository._isEmojiSupported( item, container ) )
```

IMHO it would be more readable.